### ISSUE TITLE
Gettext replace other value in dom

### ISSUE THREAD

**Issue by Matsa59** [phoenixframework/phoenix_live_view#1788]
### Environment

* Elixir version (elixir -v): 1.12.2
* Phoenix version (mix deps): 1.6.2
* Phoenix LiveView version (mix deps): 0.17.0 to 0.17.2 (included)
* Operating system: Arch Linux
* Browsers you attempted to reproduce this bug on (the more the merrier): Not related
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Not related

Let's start by code snippet in the `root.html.heex`

```elixir
<%= for value <- ["titi", "tata"] do %>
  <div>
    <%= value %>

    <%= link to: "#" do %>
      <%= gettext("Hum?") %>
    <% end %>
  </div>
<% end %>
```

### Actual behavior

```html
<div>
  Hum?
   <a href="#">Hum?</a>
</div>

<div>
  Hum?
   <a href="#">Hum?</a>
</div>
```

### Expected behavior

```html
<div>
  titi
   <a href="#">Hum?</a>
</div>
<div>
  tata
   <a href="#">Hum?</a>
</div>
```

It's not impacting me since I upgrade to 0.17.5 but it took me so many times to figure it out what happen x)

I can't find any issue referring to this bug, sorry if someone already mentioned it.

**Comment by josevalim** [comment]
Sorry, do you mean the bug no longer happens on v0.17.5?

**Comment by Matsa59** [comment]
Yes it's fixed in 0.17.3 but I suggest to warn people using older version that they will have side effect when using `gettext/1` macro

**Comment by josevalim** [comment]
Oh, it is not related to gettext. It is related to rendering .heex templates in certain cases. We have fixed most edge cases but there are one or two still recently reported. See #1783. :)
